src/services/users.service.ts

```diff
@@ -52,6 +52,17 @@ class UserService {
     const deleteUserData = await this.users.delete({ where: { id: userId }, select: { email: true, id: true } });
     return deleteUserData;
   }
+
+  public async manageGroups(userId: number, data: any) {
+    if (isEmpty(data)) throw new HttpException(400, "Groups don't exist");
+    
+    const groups = await this.users.update({
+      where: { id: userId },
+      data
+    })
+
+    return groups;
+  }
```

Comment:
There is a bit of ambiguity as to the purpose of this function, my assumption is that it takes in a set of groupIds, and then assigns the users to all those groups. 

```suggestion
  public async manageGroups(userId: number, data: number[]) {
    if (isEmpty(data)) throw new HttpException(400, "Groups don't exist");
    
    await this.member.deleteMany({ where: { user_id: userId } });
    
    await prisma.member.createMany({
          data: data.map(groupId => ({
            user_id: userId,
            group_id: groupId,
          })),
          skipDuplicates: true, // Prevents errors if a membership already exists
        });
    
    const userGroups = await this.users.findUnique({
      where: { id: userId },
      select: {
        groups: {
          select: {
            group: { select: { id: true, name: true } },
          },
        },
      },
    });

    return userGroups?.groups.map(member => member.group) || [];
  }
```